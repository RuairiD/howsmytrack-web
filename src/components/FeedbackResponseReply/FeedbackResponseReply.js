import React from 'react';
import dateFormat from 'dateformat';

import { Comment } from 'antd';

export type FeedbackResponseReplyProps = {
    username: string,
    text: string,
    timeCreated: string,
};

class FeedbackResponseReply extends React.Component<FeedbackResponseReplyProps> {
    /*
     * Component for showing a reply to some feedback as part of a thread.
     */
    render() {
        return (
            <Comment
                author={this.props.username}
                content={
                    <p>{this.props.text}</p>
                }
                datetime={dateFormat(
                    new Date(
                        Date.parse(this.props.timeCreated)
                    ),
                    'mmmm dS yyyy h:MM TT',
                )}
            />
        );
    }
}

export default FeedbackResponseReply;
