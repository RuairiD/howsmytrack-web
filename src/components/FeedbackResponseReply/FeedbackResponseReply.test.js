import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import FeedbackResponseReply from "./FeedbackResponseReply";

describe("FeedbackResponseReply", () => {
    it("renders a Comment containing the reply's content", () => {
        const wrapper = shallow(
            <FeedbackResponseReply
                username="username"
                text="text"
                timeCreated={new Date(1405274400000)}
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
