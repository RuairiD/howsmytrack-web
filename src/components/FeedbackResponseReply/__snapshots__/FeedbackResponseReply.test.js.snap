// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackResponseReply renders a Comment containing the reply's content 1`] = `
<Comment
  author="username"
  content="text"
  datetime="July 13th 2014 11:00 AM"
/>
`;
